server/src/main/java/org/elasticsearch/search/suggest/phrase/DirectCandidateGenerator.java

```diff
@@ -132,31 +132,41 @@ public String getField() {
     public CandidateSet drawCandidates(CandidateSet set) throws IOException {
         Candidate original = set.originalTerm;
         BytesRef term = preFilter(original.term, spare, byteSpare);
-        if (suggestMode != SuggestMode.SUGGEST_ALWAYS) {
-            /**
-             * We use the {@link TermStats#docFreq} to compute the frequency threshold
-             * because that's what {@link DirectSpellChecker#suggestSimilar} expects
-             * when filtering terms.
-             */
-            int threshold = thresholdTermFrequency(original.termStats.docFreq);
-            if (threshold == Integer.MAX_VALUE) {
-                // the threshold is the max possible frequency so we can skip the search
-                return set;
+        float origThreshold = spellchecker.getThresholdFrequency();
+        try {
+            if (suggestMode != SuggestMode.SUGGEST_ALWAYS) {
+                /**
+                 * We use the {@link TermStats#docFreq} to compute the frequency threshold
+                 * because that's what {@link DirectSpellChecker#suggestSimilar} expects
+                 * when filtering terms.
+                 */
+                int threshold = thresholdTermFrequency(original.termStats.docFreq);
+                if (threshold == Integer.MAX_VALUE) {
+                    // the threshold is the max possible frequency so we can skip the search
+                    return set;
+                }
+                // don't override the threshold if the provided min_doc_freq is greater
+                // than the original term frequency.
+                if (spellchecker.getThresholdFrequency() < threshold) {
+                    spellchecker.setThresholdFrequency(threshold);
+                }
             }
-            spellchecker.setThresholdFrequency(threshold);
```

Comment:
Reading this old code: since this part changed the spellchecker configuration permanently (without resetting its), maybe we don't even need the "finally" clause to change it back? It looks like the spellchecker is only used with this threshhold afterwards anyway?

Reply:
No, see the comment above